src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -356,6 +361,7 @@ protected void install(InstallConfig installConfig) {
 
                         setupSync(newSyncDirection, Optional.of(currentDirection));
                         syncHandler.setSyncDirection(newSyncDirection);
+                        merger.setSyncDirection(newSyncDirection);
```

Comment:
could you have the RequestMerger in the `SyncHandler` so you could set the direction for the merger in `syncHandler.setSyncDirection`?